corehq/apps/export/tasks.py

```diff
@@ -24,27 +24,29 @@
 
 @task(queue='export_download_queue')
 def populate_export_download_task(export_instances, filters, download_id, filename=None, expiry=10 * 60 * 60):
-    export_file = get_export_file(
-        export_instances,
-        filters,
-        # We don't have a great way to calculate progress if it's a bulk download,
-        # so only track the progress for single instance exports.
-        progress_tracker=populate_export_download_task if len(export_instances) == 1 else None
-    )
+    with TransientTempfile() as temp_path:
+        export_file = get_export_file(
+            export_instances,
+            filters,
+            temp_path,
+            # We don't have a great way to calculate progress if it's a bulk download,
+            # so only track the progress for single instance exports.
+            progress_tracker=populate_export_download_task if len(export_instances) == 1 else None
+        )
 
-    file_format = Format.from_format(export_file.format)
-    filename = filename or export_instances[0].name
+        file_format = Format.from_format(export_file.format)
+        filename = filename or export_instances[0].name
 
-    with export_file as file_:
-        db = get_blob_db()
-        db.put(file_, download_id, timeout=expiry)
+        with export_file as file_:
+            db = get_blob_db()
+            db.put(file_, download_id, timeout=expiry)
 
-        expose_blob_download(
-            download_id,
-            mimetype=file_format.mimetype,
-            content_disposition=safe_filename_header(filename, file_format.extension),
-            download_id=download_id,
-        )
+            expose_blob_download(
+                download_id,
+                mimetype=file_format.mimetype,
+                content_disposition=safe_filename_header(filename, file_format.extension),
+                download_id=download_id,
+            )
```

Comment:
Can use `ExposeBlobDownload` here to do some of this? Probably out of scope of this PR so feel free to skip it if needed.